### PR TITLE
.NET Maui "Single Project"

### PR DESCRIPTION
**Pull request by jonathanpeppers**
### Description of Change ###

Context: https://github.com/xamarin/xamarin-android/blob/fcd9c30164d4ac0e88440357b5c96b9fdebccb39/Documentation/guides/OneDotNetSingleProject.md

We are aiming to get a new project type working that would have a
`.csproj` such as:

    <Project Sdk="Microsoft.NET.Sdk">
      <PropertyGroup>
        <TargetFramework>net6.0-android;net6.0-ios</TargetFramework>
        <OutputType>Exe</OutputType>
        <SingleProject>true</SingleProject>
      </PropertyGroup>
    </Project>

Within the project you would have:

* `*.cs` all shared code
* `iOS/*` - iOS only stuff, including `Info.plist`
* `Android/*` - Android only stuff, including `AndroidManifest.xml`

I also used some other new settings that results in a smaller
`Info.plist` and `AndroidManifest.xml`:

    <ApplicationTitle>Single Project</ApplicationTitle>
    <ApplicationId>com.microsoft.maui.singleproject</ApplicationId>
    <ApplicationVersion>1.0</ApplicationVersion>
    <AndroidVersionCode>1</AndroidVersionCode>

The eventual goal being we might be able to completely get rid of
`Info.plist` and `AndroidManifest.xml` for simple apps. You would only
need them for edge-case, platform-specific features.

To get this working within a NuGet package, we can include two new
files:

* `Xamarin.Forms.SingleProject.props`
* `Xamarin.Forms.SingleProject.targets`

They only enable things conditionally, if `$(SingleProject)` is set.

I created a new sample with a single `Label` to demonstrate things
working.

### API Changes ###

None

### Platforms Affected ### 

- Core/MSBuild (all platforms)

### Behavioral/Visual Changes ###

A new `$(SingleProject)` MSBuild property will opt you into a new, magical project type.

### Before/After Screenshots ### 

Not applicable

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: .nuspec/Xamarin.Forms.SingleProject.props [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<!-- "Single Project" support - all <ItemGroup/>'s must check if $(EnableDefaultItems) and $(SingleProject) are true -->
+<Project>
```
Comment (jonathanpeppers) [inline .nuspec/Xamarin.Forms.SingleProject.props:2]: Do I need to name this `Microsoft.Maui.SingleProject.props/targets`?

Comment (rmarinho) [reply .nuspec/Xamarin.Forms.SingleProject.props:2]: Microsoft.Maui.Controls.SingleProject

Comment (jonathanpeppers) [reply .nuspec/Xamarin.Forms.SingleProject.props:2]: This is the similar to `Xamarin.Forms.props`, does it really need `.Controls` in the name?

Comment (jonathanpeppers) [reply .nuspec/Xamarin.Forms.SingleProject.props:2]: Ok, I guess it is: https://github.com/xamarin/Xamarin.Forms/tree/main-handler/.nuspec

**Review by rmarinho — APPROVED** [review]

**Review by jsuarezruiz — APPROVED** [review]

**Review by PureWeen — APPROVED** [review]

**Pull request merged**